libs/npm/core/src/components/formik/sentiment/FormikSentiment.tsx

```diff
@@ -64,40 +75,65 @@ export const FormikSentiment = <T extends object>({
     if (active !== value) {
       setActive(value);
     }
-  }, [active, value]);
+  }, [value, active]);
 
   const determineIndicator = (option: number) => {
     if (option === 5) {
-      return <img alt={option.toString()} src={FaceGrinWide} />;
+      return coloredIcon ? (
+        <FaSmile className="tone-icon" color="#59E9BE" />
+      ) : (
+        <img alt={option.toString()} src={FaceGrinWide} />
+      );
     } else if (option === 0) {
-      return <img alt={option.toString()} src={FaceMeh} />;
+      return coloredIcon ? (
+        <FaMeh className="tone-icon" color="#F1C02D" />
+      ) : (
+        <img alt={option.toString()} src={FaceMeh} />
+      );
     } else if (option === -5) {
-      return <img alt={option.toString()} src={FaceFrownOpen} />;
+      return coloredIcon ? (
+        <FaFrown className="tone-icon" color="#EB8585" />
+      ) : (
+        <img alt={option.toString()} src={FaceFrownOpen} />
+      );
     } else {
       return <span className="blank">&nbsp;</span>;
     }
   };
 
+  const handleOptionClick = (option: number) => {
+    if (!defaultTonePool) return;
+
+    setActive(option);
+
+    const existingTonePoolIndex = tonePools?.findIndex((pool) => pool.id === defaultTonePool.id);
+    let updatedTonePools;
+
+    if (existingTonePoolIndex !== undefined && existingTonePoolIndex >= 0) {
+      // Replace the existing TonePool value
+      updatedTonePools = tonePools?.map((pool, index) =>
+        index === existingTonePoolIndex ? { ...pool, value: option } : pool,
+      );
+    } else {
+      // Add a new TonePool entry
+      updatedTonePools = [...(tonePools || []), { ...defaultTonePool, value: option }];
+    }
+
+    setFieldValue(name.toString(), updatedTonePools, true);
```

Comment:
Having both of these can result in odd state.  Formik will update the form context and then another separate set of state is being updated next.  It is easier to maintain if you make the `setFieldValue` work correctly.

Reply:
Not using FormikSentiment.